src/binary-reader-ir.cc

```diff
@@ -1255,6 +1255,10 @@ Result BinaryReaderIR::OnDataSymbol(Index index, uint32_t flags,
   if (name.empty()) {
     return Result::Ok;
   }
+  if (flags & WABT_SYMBOL_FLAG_UNDEFINED) {
```

Comment:
Is there a simple test we can add for this?

Reply:
Would need to be a gen-wasm-based test since we can't generate these symbols from wat files (yet).  

Maybe just modifying test/binary/linking-section.txt would be enough?

Reply:
It can't be `test/binary/linking-section.txt` since it has a data section, the test was triggered by the combination of a lack of data section and an undefined data symbol.

Reply:
btw should probably add to the waterfall to run the decompiler on all torture tests :)

Reply:
sgtm!

Reply:
Given that its already covered by the torture tests and this not likely to regress, I don't think a new test is needed.